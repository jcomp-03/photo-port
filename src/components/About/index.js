import React from 'react';

const About = () => {
    return (
        <section>
            <h1>Who am I?</h1>
        </section>
    );
}

export default About;