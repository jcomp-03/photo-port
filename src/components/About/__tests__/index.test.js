import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import About from '..';


// we'll call the cleanup function using the afterEach
// global function from Jest
afterEach(cleanup);

// use the describe function to declare the component
// we're testing
describe('Testing the \'About\' component', () => {
    //renders About test
    // First test; we use the it function. First argument is a
    // string declaring what is being tested; in second argument
    // a callback function runs the test
    it('renders', () => {
        render(<About />);
    });

    // Second test
    it('matches snapshot DOM node structure', () => {
        // render About
        const { asFragment } = render(<About />);
        expect(asFragment()).toMatchSnapshot();
    })
})