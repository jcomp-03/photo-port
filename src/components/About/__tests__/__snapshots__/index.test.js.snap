// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the 'About' component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="my-5"
  >
    <h1>
      Who am I? A figment of your imagination.
    </h1>
    <img
      alt="cover"
      class="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;
